### DISCUSSION TITLE
SaCredential on Verbose

### DISCUSSION THREAD

**Discussion by jarias-korewireless** [dataplat/dbatools#8049] [Contributors]
https://github.com/sqlcollaborative/dbatools/blob/c817c4e80256dccaa17cd5d451196d3558772248/internal/functions/Invoke-Program.ps1#L167

How about adding some sort of mask to cover the Sa password on Verbose message?

**$secureArgumentList = $ArgumentList.replace($sacredential.GetNetworkCredential().Password,"****")**
Write-Message -Level Verbose -Message "Starting process [$Path] with arguments **[$secureArgumentList]** on $ComputerName through $Authentication protocol"

**Comment by potatoqualitee** [comment]
Good call. Lez see what I can do real quick.

**Reply by potatoqualitee** [reply]
I've submitted https://github.com/dataplat/dbatools/pull/8095 but do not have time to test it, though the code looks reasonable. I'll have to get a confirmation before merge.
